File: dagger/app/src/main/java/com/dagger/mvp/MVPApi.java
package com.dagger.mvp;


import com.dagger.mvp.model.MVPBean;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * by y on 2017/5/31.
 */

public class MVPApi {
    public static final String ZL_BASE_API = "https://zhuanlan.zhihu.com/api/";

    public interface ZLService {
        @GET("columns/" + "{suffix}/posts")
        Observable<List<MVPBean>> getList(@Path("suffix") String suffix, @Query("limit") int limit, @Query("offset") int offset);
    }
}

File: UtilsSample/Dagger/src/main/java/sample/util/develop/android/dagger/mvp/MVPApi.kt
package sample.util.develop.android.dagger.mvp


import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import sample.util.develop.android.dagger.mvp.model.MVPBean

/**
 * by y on 2017/5/31.
 */

object MVPApi {
    const val ZL_BASE_API = "https://zhuanlan.zhihu.com/api/"

    interface ZLService {
        @GET("columns/" + "{suffix}/posts")
        fun getList(@Path("suffix") suffix: String, @Query("limit") limit: Int, @Query("offset") offset: Int): Observable<List<MVPBean>>
    }
}
